agent/engine/docker_task_engine.go

```diff
@@ -873,6 +873,12 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 		return metadata
 	}
 	pullSucceeded := metadata.Error == nil
+	if pullSucceeded {
+		dockerContainer := &apicontainer.DockerContainer{
+			Container: container,
+		}
+		engine.state.AddContainer(dockerContainer, task)
```

Comment:
Should we do this only when the config var is set?

Reply:
The cfg var check is added on line 876 in [this commit](https://github.com/aws/amazon-ecs-agent/pull/2700/commits/fba2d68cb35ed883fb1b920417c02caa77a43981). Thank you.